integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -387,6 +389,71 @@ fn next_block_has_new_shard_layout(epoch_manager: Arc<dyn EpochManagerAdapter>,
         && shard_layout != next_shard_layout
 }
 
+/// Asserts that for each child shard (Mem)Trie and FlatState contains the same final state.
+fn assert_state_sanity_for_children_shard(parent_shard_uid: ShardUId, client: &Client) {
+    let final_head = client.chain.final_head().unwrap();
```

Comment:
Can this check be done at any height?

Reply:
Not any height, because we know new shards' flat state won't be available at the first block of the new epoch. It could be done for any block after resharding post processing is finished, but I feel it's a bit too complicated for the value added.